src/types/frozenset.rs

```diff
@@ -31,9 +34,11 @@ impl PyFrozenSet {
     /// Creates a new frozenset.
     ///
     /// May panic when running out of memory.
-    pub fn new<'p, T: ToPyObject>(py: Python<'p>, elements: &[T]) -> PyResult<&'p PyFrozenSet> {
-        let list = elements.to_object(py);
-        unsafe { py.from_owned_ptr_or_err(ffi::PyFrozenSet_New(list.as_ptr())) }
+    pub fn new<'p, T: ToPyObject>(
```

Comment:
Just for the record, this is breaking since `&[T]` is `IntoIterator<Item = &T>`.

Reply:
Hmm. I think in which case let's use `&T` here for now too. We can revisit that as a different breaking change in future.

Reply:
If we're going to be breaking code anyway, what about taking `IntoPy` instead? i.e. inserting pyclasses into the set by value?

Reply:
Yes, I think I'd like to do this (and I've thought similar in the past for `PyList::new`). I will hold off from doing that here, as this is again non-breaking after I've corrected to use `Item = &T` for now.

I'd be keen to revisit this along with #2316 and also a design about how we maybe also handle fallible into-py conversions.